src/libraries/System.Net.WebSockets.Client/tests/SendReceiveTest.cs

```diff
@@ -12,36 +12,130 @@
 
 namespace System.Net.WebSockets.Client.Tests
 {
+    // Memory array segment----------------
 
-    public sealed class InvokerMemorySendReceiveTest : MemorySendReceiveTest
+    // Invoker--------
+
+    // Local----
+
+    [SkipOnPlatform(TestPlatforms.Browser, "Custom invoker is ignored on Browser")]
+    public class InvokerMemorySendReceiveLocalTest : InvokerMemorySendReceiveTest
+    {
+        public InvokerMemorySendReceiveLocalTest(ITestOutputHelper output) : base(output) { }
+    }
+
+    [SkipOnPlatform(TestPlatforms.Browser, "Custom invoker is ignored on Browser")]
+    public class InvokerMemorySendReceiveLocalSslTest : InvokerMemorySendReceiveTest
+    {
+        public InvokerMemorySendReceiveLocalSslTest(ITestOutputHelper output) : base(output) { }
+        protected override bool UseSsl => true;
+    }
+
+    // External server -- HTTP/1.1 only----
+    // TODO: add HTTP/2 when https://corefx-net-http2.azurewebsites.net will support extended connect
+
+    [OuterLoop("Uses external servers", typeof(PlatformDetection), nameof(PlatformDetection.LocalEchoServerIsNotAvailable))]
+    [SkipOnPlatform(TestPlatforms.Browser, "Custom invoker is ignored on Browser")]
+    public sealed class InvokerMemorySendReceiveRemoteSslTest : InvokerMemorySendReceiveRemoteTest
+    {
+        public InvokerMemorySendReceiveRemoteSslTest(ITestOutputHelper output) : base(output) { }
+        protected override bool UseSsl => true;
+    }
+
+    [OuterLoop("Uses external servers", typeof(PlatformDetection), nameof(PlatformDetection.LocalEchoServerIsNotAvailable))]
+    [SkipOnPlatform(TestPlatforms.Browser, "Custom invoker is ignored on Browser")]
+    public class InvokerMemorySendReceiveRemoteTest : InvokerMemorySendReceiveTest
+    {
+        public InvokerMemorySendReceiveRemoteTest(ITestOutputHelper output) : base(output) { }
+
+        protected override bool UseRemoteServer => true;
+    }
+
+    public abstract class InvokerMemorySendReceiveTest : MemorySendReceiveTest
     {
         public InvokerMemorySendReceiveTest(ITestOutputHelper output) : base(output) { }
 
         protected override bool UseCustomInvoker => true;
     }
 
-    public sealed class HttpClientMemorySendReceiveTest : MemorySendReceiveTest
+    //HttpClient--------
+
+    // Local----
+
+    [SkipOnPlatform(TestPlatforms.Browser, "Custom invoker is ignored on Browser")]
+    public class HttpClientMemorySendReceiveLocalTest : HttpClientMemorySendReceiveTest
```

Comment:
If we are not going to update all SendReceive tests to new format in this PR, then I'm concerned that all old-format tests would run duplicatively multiple times for each of the child classes. This seems like excessive use of CI time without adding much value...

Reply:
Condition added